### PR TITLE
augmentation with no generated tests

### PR DESCRIPTION
**Pull request by haim-kermany**
## Description

supporting the case of augmenting when there is no ctd tests generated
with and without dev_tests
Related to # 90

## Type of Change

Please check the types of changes your PR introduces.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Refactoring (non-breaking code restructuring that preserves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build-related update (CI workflow, test cases)
- [ ] Documentation update
- [ ] Other (please describe):

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so that it can be replicated.
Please also list any relevant details for your test configuration.
new unitest  - test_augment_on_empty_test_suite() that runs augment_with_code_coverage() twice. 
- [ ] Test A
- [ ] Test B

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/data/failing/src/main/java/failing/Failing.java [hunk 1]
```diff
@@ -13,10 +13,4 @@ public void throwCaughtException(int input) throws IOException {
 		throw new IOException("Throwing IOException");
 	}
 
-	public void throwUncaughtException(String input) throws NullPointerException {
```
Comment (rachelt44) [inline test/data/failing/src/main/java/failing/Failing.java:16]: can you add a new file instead of splitting the existing one to two files?

File: test/data/failing/tkltest_config.toml [hunk 2]
```diff
@@ -1,57 +1,56 @@
...
-refactored_app_path_prefix = ""
-refactored_app_path_suffix = []
-reuse_base_tests = false
-
-[generate.evosuite]
-criterion = [ "BRANCH",]
-
-[generate.randoop]
-no_error_revealing_tests = false
+[general]
```
Comment (rachelt44) [inline test/data/failing/tkltest_config.toml:1]: why was the config file changed?

File: test/data/failing/user-tests/build.gradle [hunk 3]
```diff
@@ -0,0 +1,139 @@
+
```
Comment (rachelt44) [inline test/data/failing/user-tests/build.gradle:1]: why use gradle file for the user tests? better to use maven like the application itself

File: test/unit/test_generate_execute.py [hunk 4]
```diff
@@ -47,7 +47,7 @@ class GenerateExecuteTest(unittest.TestCase):
             'test_directory': '__splitNjoin-generated-tests',
         },
         'failing' : {
-            'config_file': os.path.join(test_data_dir, 'failingApp', 'tkltest_config.toml'),
+            'config_file': os.path.join(test_data_dir, 'failing', 'tkltest_config.toml'),
```
Comment (rachelt44) [inline test/unit/test_generate_execute.py:50]: why did you change the name of the application?

**Review by rachelt44 — CHANGES_REQUESTED** [review]

File: test/data/failing/tkltest_config.toml (hunk 2 above)
Comment (haim-kermany) [reply test/data/failing/tkltest_config.toml:1]: 1. add dev_test
2. the changed the directory from failingApp to failing, so the app_path and the class path was changed

File: test/data/failing/user-tests/build.gradle (hunk 3 above)
Comment (haim-kermany) [reply test/data/failing/user-tests/build.gradle:1]: it is easy for me to implement gradle

File: test/unit/test_generate_execute.py (hunk 4 above)
Comment (haim-kermany) [reply test/unit/test_generate_execute.py:50]: there was a convention that the app name= app dir.

File: test/unit/test_unit_tests.py [hunk 5]
```diff
@@ -418,8 +424,62 @@ def test_exclude_classes_covered_by_dev_test(self) -> None:
...
-            self.assertTrue(config['generate']['excluded_class_list'] == test_apps[app_name]['covered_classes'])
+            self.assertTrue(set(config['generate']['excluded_class_list']) == set(test_apps[app_name]['covered_classes']))
+
+        self.__assert_no_artifact_at_cli(test_apps.keys())
+
+    def test_augment_on_empty_test_suite(self) -> None:
+        """Test that we can augment on an empty ctd test suite"""
+        test_apps = self.bad_path_test_apps
 
+        for app_name in test_apps.keys():
```
Comment (rachelt44) [inline test/unit/test_unit_tests.py:435]: it is redundant to test the empty test suite case on all apps. One app is enough.

**Review by rachelt44 — CHANGES_REQUESTED** [review]

File: test/unit/test_unit_tests.py (hunk 5 above)
Comment (haim-kermany) [reply test/unit/test_unit_tests.py:435]: not sure I understand the comment, only one app is tested:
        **test_apps = self.bad_path_test_apps
        for app_name in test_apps.keys():
            dir_util.cd_cli_dir()**

Comment (rachelt44) [reply test/unit/test_unit_tests.py:435]: Right, missed that. But better to get the app directly by its name and not iterate over bad paths test apps, because if we add more bad path apps to this test file, we don't want them to be automatically added to this test method. We need just a single corner case.

**Review by rachelt44 — APPROVED** [review]
